spyder/api/asyncdispatcher.py

```diff
@@ -181,28 +190,28 @@ async def my_coroutine(...):
 
     @typing.overload
     def __call__(
-        self: AsyncDispatcher[typing.Awaitable[_T]],
-        async_func: typing.Callable[_P, typing.Awaitable[_T]],
-    ) -> typing.Callable[_P, typing.Awaitable[_T]]: ...
+        self: AsyncDispatcher[collections.abc.Awaitable[_T]],
+        async_func: typing.Callable[_P, collections.abc.Awaitable[_T]],
```

Comment:
`typing.Callable` can be replaced by `collections.abc.Callable` since Python 3.9. So please change it in all places it's used in this file.

Reply:
Right; my understanding reading that docs for `typing.Callable` was that support for `ParamSpec` in `collections.abc.Callable` was only added in 3.10, so I'd changed it elsewhere but not here where `ParamSpec` was used. However, reviewing it further it seems I misread the note and in fact `typing.Callable` doesn't support `ParamSpec` either (as it was only added in 3.10 to begin with), so it doesn't actually make a difference. We will need to either use `Callable` from `typing_extensions `, or just take advantage `from __future__ import annotations` and use the `collections.abc` version since an issue would only present itself attempting to use annotations at runtime on 3.9 (which doesn't happen, possibly outside of the docs build).